src/Downloader/Args.cs

```diff
@@ -0,0 +1,226 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+public struct Args
+{
+    public string Org { get; set; }
+    public List<string> Repos { get; set; }
+    public string GithubToken { get; set; }
+    public string? IssueDataPath { get; set; }
+    public int? IssueLimit { get; set; }
+    public string? PullDataPath { get; set; }
+    public int? PullLimit { get; set; }
+    public int? PageSize { get; set; }
+    public int? PageLimit { get; set; }
+    public int[] Retries { get; set; }
+    public Predicate<string> LabelPredicate { get; set; }
+    public bool Verbose { get; set; }
+
+    static void ShowUsage(string? message = null)
+    {
+        string executableName = Process.GetCurrentProcess().ProcessName;
+
+        Console.WriteLine($$"""
+            ERROR: Invalid or missing arguments.{{(message is null ? "" : " " + message)}}
+
+            Usage:
+              {{executableName}} --repo {org/repo1}[,{org/repo2},...] --label-prefix {label-prefix} [options]
+
+              Required arguments:
+                --repo              The GitHub repositories in format org/repo (comma separated for multiple).
+                --label-prefix      Prefix for label predictions.
+
+              Required for downloading issue data:
+                --issue-data        Path for issue data file to create (TSV file).
+
+              Required for downloading pull request data:
+                --pull-data         Path for pull request data file to create (TSV file).
+
+              Optional arguments:
+                --issue-limit       Maximum number of issues to download.
+                --pull-limit        Maximum number of pull requests to download.
+                --page-size         Number of items per page in GitHub API requests.
+                --page-limit        Maximum number of pages to retrieve.
+                --retries           Comma-separated retry delays in seconds. Default: 30,30,300,300,3000,3000.
+                --token             GitHub access token. Default: Read from GITHUB_TOKEN env var.
+                --verbose           Enable verbose output.
```

Comment:
[nit] GHC was giving me a similar suggestion, but I found it rather unclear - that was the reason to list several usage patterns (where applicable).
This is not a blocking comment, but something to consider nevertheless.

In the specific case of the Downloader the required args are only `--repo {{org/repo1}}[,{{org/repo2}},...] --label-prefix {{label-prefix}}`. All other args are optional.

Reply:
They're _sort of_ optional. You have to specify at least one of `--issue-data` or `--pull-data`. But there's no perfect way to express it. And... good enough is good enough.